lib/src/http_client_request.dart

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+// import 'dart:collection';
+import 'dart:convert';
+import 'dart:developer';
+import 'dart:ffi';
+import 'dart:io';
+import 'dart:isolate';
+import 'dart:typed_data';
+
+import 'package:ffi/ffi.dart';
+
+import 'exceptions.dart';
+
+import 'generated_bindings.dart';
+
+// TODO: Enable these when Headers are implemented
+// part '../third_party/http_headers.dart';
+// part '../third_party/http_date.dart';
+part 'http_client_response.dart';
+part 'http_callback_handler.dart';
+
+// Type definitions for various callbacks
+
+/// Called when a redirect is received. Gets new location and response code as
```

Comment:
Typedefs never get called. Maybe:

`Function signature for callbacks called on redirects.`

You can omit the argument description, because they are already in the typedef.

(Same all the other typedefs.)